src/main/java/org/opentripplanner/middleware/controllers/api/ApiUserController.java

```diff
@@ -24,8 +29,32 @@ protected ApiUser getUserProfile(Auth0UserProfile profile) {
      */
     @Override
     ApiUser preCreateHook(ApiUser user, Request req) {
-        // TODO: Add the API keys for this user.
+        String apiKeyId = AwsApiGateway.getApiKeyId(user.id);
+        if (apiKeyId == null) {
+            logMessageAndHalt(req, HttpStatus.INTERNAL_SERVER_ERROR_500,
+                String.format("Unable to get AWS api key for user %s", user)
+                , null);
+        }
+
+        //FIXME This suggests many, where only one is returned?
+        user.apiKeyIds.add(apiKeyId);
```

Comment:
I don't think we've entirely figured out how the API keys will correspond to the various endpoints in the API, but we want to leave open the possibility that more than one API key can be given to an API user.

Reply:
Have removed comment.